collector/filesystem_common.go

```diff
@@ -126,7 +127,15 @@ func (c *filesystemCollector) Update(ch chan<- prometheus.Metric) (err error) {
 	if err != nil {
 		return err
 	}
+	// Make sure we expose a metric once, even if there are multiple mounts
+	updated := map[string]bool{}
 	for _, s := range stats {
+		k := fmt.Sprintf("%q", s)
```

Comment:
Here is the the more accurate description of what types can be compared and used as map keys: https://golang.org/ref/spec#Comparison_operators